src/build_and_run_task.cr

```diff
@@ -0,0 +1,60 @@
+# Builds and runs a file (typically tasks.cr) with the passed in args
+class LuckyCli::BuildAndRunTask
+  private getter tasks_file : String
+  private getter args : String
+  private getter fake_stdout = IO::Memory.new
+  private getter fake_stderr = IO::Memory.new
+  private getter tasks_binary_path = "tmp/tasks_binary"
+
+  def initialize(@tasks_file, @args)
+  end
+
+  def self.call(*args, **named_args)
+    new(*args, **named_args).call
+  end
+
+  def call
+    FileUtils.mkdir_p("tmp")
+    build_status = build_tasks_binary
```

Comment:
I wonder if we could do some sort of cache here. Maybe compare md5 or something so we don't have to build every time. 🤔  That could be a future thing, but that would be awesome if the second time you ran a task it was a lot faster.

Reply:
Yes I think so. I was thinking about how to do it and I think we'd need to extract/copy some of the code in lucky watch that checks the timestamps. Then we'd compare the timestamps of the tasks binary and the last modifification date of the built files, but I'm not sure. Would definitely be cool so if you run a few tasks in a row you would only compile once!

I'll do it in a separate change at some point though since caching stuff is known to be quite tricky :P

Reply:
Yeah, a separate PR for sure. This is already quite a bit. But I just had an idea. Before compiling, we just take the file before we build it, and get the sha and/or timestamp of it, and then create a lucky cache directory. Then we just check the cache and skip building. There's already a `~/.cache/crystal/` and `~/.cache/shards/`. We might as well just hook in to that.

Reply:
@jwoertink that is a cool idea. I was thinking of how it’d work but I’m not sure it will. We could cache the sha of the tasks file, but the tasks file depends on a bunch of other files and folders, like models, tasks from shards, etc so it’d have to track those too. So we’d have to sha everything which I don’t think would work. 

Even if we could sha everything  I think it may be simpler to do a last modified traversal where we give it a list of files and folders and it gets the last modified file or folder and compares with the last modified timestamp on the binary. I think that’d work since if any files change, like a model, it’ll recompile the tasks binary. Does that make sense or is there another way to approach this?